### PR TITLE
Prepare nc

### PR DESCRIPTION
**Pull request by hannes14**
prepare for Plural rings.
Simplest cases: rWeyl,rExterior: like rDefault for commutative rings

### CODE REVIEW

**Comment by fieker** [comment]
apart from that I don't mind, is there any reason to not print the Singular.jl version number as well?

**Comment by mohamed-barakat** [comment]
I am afraid I will need some time to test this PR with homalg in order to give substantial feedback. I need at least two weeks from now. This should not stop anyone from merging if needed by someone urgently.

**Comment by hannes14** [comment]
It only adds functionality (and is not complete: only the preparation to define nc(plural) rings.)
Should not conflict with anything else.

File: src/Singular.jl [hunk 1]
```diff
@@ -116,8 +116,8 @@ function __init__()
    if show_banner             
      println("""Singular.jl, based on
                      SINGULAR                                 /  
- A Computer Algebra System for Polynomial Computations       /  Singular.jl: $VERSION_NUMBER 
-                                                           0<   Singular   : $svn
+ A Computer Algebra System for Polynomial Computations       /   Singular.jl: $VERSION_NUMBER
+                                                           0<    Singular: $svn
```
Comment (fingolfin) [inline src/Singular.jl:120]: This is an unrelated change?

File: src/libsingular/rings.jl [hunk 2]
```diff
@@ -6,6 +6,22 @@ function rDefault(cf::coeffs_ptr, vars::Array{T,1}, ord::Array{rRingOrder_t, 1},
...
+   return r
+end
+
+function rExterior(cf::coeffs_ptr, vars::Array{T,1}, ord::Array{rRingOrder_t, 1},
+   blk0::Array{Cint, 1}, blk1::Array{Cint, 1}, bitmask::Culong) where {T}
+   blk0ptr = pointer(blk0)
+   blk1ptr = pointer(blk1)
+   r = rDefault_Exterior_helper(cf, vars, ord, blk0ptr, blk1ptr, bitmask);
+   return r
+end
```
Comment (fingolfin) [inline src/libsingular/rings.jl:23]: It concerns me a bit that we have zero tests for this. It would be good to at least invoke them once, to make sure they don't do something silly, like crashing immediately..

File: deps/src/rings.cpp [hunk 3]
```diff
@@ -43,13 +43,84 @@ auto rDefault_long_helper(coeffs                        cf,
...
+        (rRingOrder_t *)omAlloc0(len_ord * sizeof(rRingOrder_t));
+    for (int i = 0; i < len_ord; i++)
+    {
+        ord_ptr[i] = ord[i];
+    }
+    int ** wvhdl = NULL;
+    auto r = rDefault(cf, len, vars_ptr, len_ord, ord_ptr, blk0, blk1, wvhdl,
+                      bitmask);
+    delete[] vars_ptr;
+    r->ShortOut = 0;
```
Comment (fingolfin) [inline deps/src/rings.cpp:70]:
```suggestion
    r = rDefault_long_helper(cf, vars, ord, blk0, blk1, bitmask);
```

File: deps/src/rings.cpp [hunk 4]
```diff
@@ -43,13 +43,84 @@ auto rDefault_long_helper(coeffs                        cf,
...
```
Comment (fingolfin) [inline deps/src/rings.cpp:104]:
```suggestion
    r = rDefault_long_helper(cf, vars, ord, blk0, blk1, bitmask);
```

**Comment by fingolfin** [comment]
@hannes14 ping?

File: src/libsingular/rings.jl (hunk 2 above)
Comment (wbhart) [reply src/libsingular/rings.jl:23]: They aren't in use yet. I'll add the high level code for this, along with tests.
